src/Servers/Kestrel/Kestrel/src/WebHostBuilderKestrelExtensions.cs

```diff
@@ -18,6 +18,22 @@ namespace Microsoft.AspNetCore.Hosting;
 /// </summary>
 public static class WebHostBuilderKestrelExtensions
 {
+    /// <summary>
+    /// TODO (acasey): doc
+    /// </summary>
+    /// <param name="hostBuilder"></param>
+    /// <returns></returns>
+    public static IWebHostBuilder UseHttpsConfiguration(this IWebHostBuilder hostBuilder)
+    {
+        HttpsConfigurationHelper.Logger.WriteLine("UseHttpsConfiguration");
+        return hostBuilder.ConfigureServices(services =>
+        {
+            // TODO (acasey): Is there a way we can just call Enable?
+            HttpsConfigurationHelper.Logger.WriteLine("UseHttpsConfiguration::ConfigureServices");
+            services.AddSingleton<HttpsConfigurationHelper.IEnabler, HttpsConfigurationHelper.Enabler>();
```

Comment:
If Enabler is registered in DI then `HttpsConfigurationHelper` ctor with IEnabler parameter is used.
If Enabler isn't registered in DI then `HttpsConfigurationHelper` parameterless ctor is used.

Do I understand that right?